### ISSUE TITLE
Can't deploy fresh function with provisionedConcurrency and maximumRetryAttempts

### ISSUE THREAD

**Issue by chrishiste** [serverless/serverless#10362]
### Are you certain it's a bug?

- [X] Yes, it looks like a bug

### Is the issue caused by a plugin?

- [X] It is not a plugin issue

### Are you using the latest version?

- [X] Yes, I'm using the latest version

### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Issue description

Hello, there is a bug that happens only when you try to deploy a fresh lambda with both a `provisionedConcurrency` and `maximumRetryAttempts` value. It seems to only happen for fresh lambdas. 

The error is the following:
`An error occurred: HelloLambdaEvConf - The function doesn't exist. (Service: AWSLambdaInternal; Status Code: 404; Error Code: ResourceNotFoundException; Request ID: a80e0c9e-4a7f-4c38-8610-93a3aaa96125; Proxy: null).`

Very easy to reproduce: 
1. `serverless create --template aws-nodejs --path myService`
2. Add `provisionedConcurrency: 1` and  `maximumRetryAttempts: 0`
3. `serverless deploy -r us-west-2`

### Service configuration (serverless.yml) content

```yaml
service: myservice

frameworkVersion: "2"

provider:
  name: aws
  runtime: nodejs12.x
  lambdaHashingVersion: 20201221
  deploymentBucket:
    name: deployment-bucket

functions:
  hello:
    handler: handler.hello
    provisionedConcurrency: 1
    maximumRetryAttempts: 0
 ```
```


### Command name and used flags

serverless deploy -r us-west-2

### Command output

```shell
Not sure I can disclose this sorry.
```

### Environment information

```shell
Framework Core: 2.68.0
Plugin: 5.5.1
SDK: 4.3.0
Components: 3.18.1
```

**Comment by medikoo** [comment]
@chrishiste great thanks for bringing this up! Indeed there's a bug. It'll be fixed with #10366

**Comment by chrishiste** [comment]
Wow I'm impressed this was fixed so quickly, thanks for the amazing work @medikoo
